rjit.c

```diff
@@ -0,0 +1,35 @@
+#include "internal/symbol.h"
+#include "ruby.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+#if USE_YJIT
+typedef void * (*intern_cstr_func)(const char *);
+
+size_t rb_rjit_address_of(void * name, intern_cstr_func func);
```

Comment:
If RJIT is being extracted as a gem, then these functions need a more generic name that doesn't include `rjit` in the string.

Reply:
The return type should probably technically be `uintptr_t`

Can you also comment to explain what the arguments to the functions are? You're passing the name, but what's the second argument for?

Reply:
> If RJIT is being extracted as a gem, then these functions need a more generic name that doesn't include `rjit` in the string.

I agree. I'll rename it.

> The return type should probably technically be `uintptr_t`

👍 


> Can you also comment to explain what the arguments to the functions are? You're passing the name to get the address, but what's the second argument for?

Yes, I'll add a comment.

Reply:
> > The return type should probably technically be `uintptr_t`
> 
> 👍

I thought that `intern_cstr_func` would be the return type at the first glance, and wondered why it took a C string.

Reply:
Shouldn't this function be `yjit` instead of `rjit`, because the function list comes from yjit?

Reply:
It should probably be just `jit_` given this is meant to be an API for external JIT gems.